### ISSUE TITLE
Is there any runnable example or document?

### ISSUE THREAD

**Issue by safeblood** [caltechads/django-ldaporm#1]
Is there any runnable example or document?

**Comment by ivanov17** [comment]
@cmalek @glennbach Sorry, are you planning to provide any documentation for this Django package?

I currently need a package that can use an LDAP directory as a database for my Django project. I was thinking about django-ldapdb, but its development has been slow lately.

What are your plans for this package? It looks like you are using this in production.

Thank you.

**Comment by glennbach** [comment]
Yes, we are using it in production in an enterprise setting, and plan to continue to use it. Development has been slow lately because it is working for our purposes. Documenting this module is unfortunately not a terribly high priority for us, but we could answer questions if you have them.

Glenn

**Comment by ivanov17** [comment]
Hello, @glennbach!

Can you please tell me if there is any way to create a relationship between a class representing an LDAP entry and another class, such as a Django user? 

In my application I also need to authenticate users using LDAP. I see that LdapManager provides authentication, but the database doesn't know anything about LDAP entries and I can't figure out how to associate the entry and the user. Could you help me please?

**Comment by cmalek** [comment]
This package is about treating LDAP as a database backend for Django, similar to the MySQL, Postgres, Oracle, etc. integrations.  It provides models and managers to behave enough like the native Django SQL ORM that the Django forms and views etc. can work with them.

Authenticating an LDAP user to a Django user is a different problem that this code does not supply, because there are already packages for that.  See https://django-auth-ldap.readthedocs.io/en/latest/

**Comment by ivanov17** [comment]
Thank you very much, Chris. This clears things up.

I initially planned to use `django-auth-ldap` package to authenticate users, but then I noticed that LdapManager can also check authentication, and decided that it would be better to ask the authors of the package about its capabilities.

Your package works great with forms and views, I'm very happy with it. Thank you.

**Comment by cmalek** [comment]
Ah, I understand now.   We have that authenticate() method on the manager class I added because in one of our use cases we can test passwords for a person.  We have several different LDAP server clusters and sometimes we need to ensure that a user has the same password in all of them.   For actual Django auth we use django-ldap-auth itself.
